doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        to release but this comes with some exceptions:
+
+        Public API only
+            The behaviour of objects accessed through private identifiers
+            (those beginning ``_``) may be changed arbitrarily between
+            versions.
+        As documented
+            We will generally assume that the users have adhered to the
+            documented parameter types and ranges. If the documentation asks
+            for a list and the user gives a tuple, we do not assure consistent
+            behavior from version to version.
+        Deprecation
+            Behaviors may change following a :term:`deprecation` period
+            (usually two releases long).  Warnings are issued using Python's
+            :mod:`warnings` module.
+        Keyword arguments
+            We may sometimes assume that all optional parameters (other than X
+            and y to :term:`fit` and similar methods) are passed as keyword
+            arguments only and may be positionally reordered.
+        Bug fixes and enhancements
+            Bug fixes and -- less often -- enhancements may change the behavior
+            of estimators, including the predictions of an estimator trained on
+            the same data and :term:`random_state`.  When this happens, we
+            attempt to note it clearly in the changelog.
+        Serialization
+            We make no assurances that pickling an estimator in one version
+            will allow it to be unpickled to an equivalent model in the
+            subsequent version.  See :ref:`persistence_limitations`.
+        :func:`utils.estimator_checks.check_estimator`
+            We provide limited backwards compatibility assurances for the
+            estimator checks: we may add extra requirements on estimators
+            covered by `estimator_checks`, usually when these were informally
+            assumed but not formally tested.
+
+        Despite this informal contract with our users, the software is provided
+        as is, as stated in the licence.  When a release inadvertently
+        introduces changes that are not backwards compatible, these are known
+        as software regressions.
+
+    categorical feature
+        A categorical or nominal :term:`feature` is one that has a finite set
+        of discrete values across the population of data.  These should be
+        represented as a column of integers. In representation, categorical
+        features are not distinguished from other integer features: ordinal or
+        count-valued. In the future we hope to support string-valued
+        categorical features. Some estimators may handle categorical features
+        better when one-hot encoded.  See the
+        `http://contrib.scikit-learn.org/categorical-encoding
+        <category_encoders>`_ package,
+        :class:`~sklearn.preprocessing.OneHotEncoder`, and a proposed
+        :issue:`9151 <CategoricalEncoder>` for tools related to encoding
+        categorical features.
+
```

Comment:
``classes_`` (required attribute of any classifier) - with really weird and inconsistent formats for multi-target :-/

Reply:
This is covered below under Attributes. Because all become `:term:` entries, we can't really duplicate entries here and there.